packages/optimizer/lib/transformers/RewriteAmpUrls.js

```diff
@@ -101,6 +107,31 @@ class RewriteAmpUrls {
     return ampUrlPrefix + url.substring(AMP_CACHE_HOST.length);
   }
 
+  _addEsm(scriptNode, preload) {
+    const esmScriptUrl = scriptNode.attribs.src.replace(/\.js$/, '.mjs');
+    if (preload) {
+      const preload = createElement('link', {
+        as: 'script',
+        crossorigin: 'anonymous',
+        href: esmScriptUrl,
+        rel: 'preload',
```

Comment:
have we considered using `modulepreload`?

Reply:
Its unsupported in Safari and Firefox :(

Reply:
is they are both included to supporting browsers try to prefetch twice?